File: sephora-backend/Application/Dtos/User/GetUserDTO.cs


using CleanArchitecture.Domain.Entities;

namespace CleanArchitecture.Application.Dtos.User;

public class GetUserDto
{
    public string Id { get; set; }
    public string UserName { get; set; } = String.Empty;
    public string Email { get; set; } = String.Empty;
    public string ProfilePicture { get; set; } = String.Empty;
    public string? PhoneNumber { get; set; }
    public List<string> Roles { get; set; }
    public DateTime RegistrationDate { get; set; }
    public ICollection<Rating>? Ratings { get; set; }
    public ICollection<Order>? Orders { get; set; }
}

File: sephora-frontend/src/models/user/GetUserDto.ts
import RatingDto from "../rating/RatingDto.ts";
import OrderDto from "../order/OrderDto.ts";
import PictureDto from "../picture/PictureDto.ts";

export default interface GetUserDto {
    get id(): string;
    get userName(): string;
    get email(): string;
    get profilePicture(): PictureDto | null;
    get phoneNumber(): string | null;
    get roles(): string[];
    get registrationDate(): Date;
    get ratings(): RatingDto[];
    get orders(): OrderDto[];
}
